enhancements/network/dual-stack.md

```diff
@@ -0,0 +1,236 @@
+  (eg, the Service `ipFamily` problem that blocked us from enabling
+  the `IPv6DualStack` feature gate in 4.4).
+
+- OVN-Kubernetes will support dual-stack networking according to the
+  model upstream Kubernetes finalizes
+
+- We enable the `IPv6DualStack` feature gate by default
+
+- Other OCP components that need to be updated to be dual-stack aware
+  will be updated. (It is not clear how much actual work is needed
+  here. We are currently researching.)
+
+- We will have CI/cluster-bot support for dual-stack clusters on
+  platforms that we support for customers (ie, bare-metal).
+
+- Assuming that the necessary CloudProvider work is completed
+  upstream, we will have CI/cluster-bot support for dual-stack
+  clusters on at least one cloud platform. (If the bare-metal solution
+  is not sufficiently useful/available, we might do some work to help
+  complete the upstream CloudProvider work.)
+
+### Non-Goals
+
+- Dual-stack support for openshift-sdn or Kuryr
+
+- Officially supporting dual-stack for third-party CNI plugins
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+#### Early CI
+
+To avoid the problems we had with single-stack IPv6, we will get CI
+set up early, to ensure that regressions do not occur in components we
+are not directly paying attention to. Note that we do not expect to do
+any work on a forked branch of OCP as we did for single-stack IPv6.
+
+Being able to bring up a dual-stack cluster at all requires enabling
+the `IPv6DualStack` feature gate in `kubelet`, `kube-apiserver`, and
+`kube-controller-manager`. We tried to do this for all clusters in 4.4
+but [ran into bugs] and had to revert it. The [upstream bugs] are not
+yet fixed.
+
+To work around this for early CI, we will patch
+`machine-config-operator`, `cluster-kube-apiserver-operator`, and
+`cluster-kube-controller-manager-operator` to add the feature gate to
+their operands if and only if the `install-config.yaml` defines a
+dual-stack cluster. (This won't allow migrating from single-stack to
+dual-stack but testing that can wait until after the upstream problems
+are fixed and we can enable the feature gate everywhere.)
+
+[ran into bugs]: https://bugzilla.redhat.com/show_bug.cgi?id=1794376
+[upstream bugs]: https://github.com/kubernetes/kubernetes/pull/86895
+
+#### Upstream Kubernetes
+
+We need to continue watching/shepherding upstream dual-stack KEPs/PRs.
+In theory, not much further work is needed upstream for dual-stack on
+bare metal. (CloudProvider support for IPv6/dual-stack is much
+poorer.)
+
+There are currently very few dual-stack-specific e2e tests upstream
+and we will probably need to create some more.
+
+#### OVN-Kubernetes
+
+Work is ongoing to add dual-stack support to OVN-Kubernetes. Upstream
+OVN-Kubernetes uses CI based on `kind` (Kubernetes-in-Docker) and we
+will have a dual-stack `kind` environment for testing it against
+upstream dual-stack Kubernetes.
```

Comment:
how about adding some links here?  You have that dual-stack tracker issue you could link, at least.